### DISCUSSION TITLE
is there a way to include a static lib to a specific project with buil_opt.h (or another mean)

### DISCUSSION THREAD

**Discussion by 0x0fe** [stm32duino/Arduino_Core_STM32#2340] [Q&A]
The only way i am aware of including a static .a library is by creating a library directory in the main arduino libraries directory, and use its library.properties file as i did below for nanoedgeai library
![image](https://github.com/stm32duino/Arduino_Core_STM32/assets/89607149/e15608b6-b943-4314-8cda-8aba5b92d7d3)

But is there a way to do this from a project directory? indeed in many case creating a global library directory is not convenient nor suitable, for example in the case of NanoEdgeAi, the library is actually tailored to the specific model and therefore will not and cannot be used for other projects, so it it is a problem to create a global library for it, plus it will cause an issue as soon as more than one project need their static library because more directories will have to be created -> mess.

**Comment by fpistm** [comment]
Hi @0x0fe 
Never tested, maybe by overriding `compiler.c.elf.extra_flags` using `platform.local.txt `or with arduino-cli command line using `--build-property`.
https://arduino.github.io/arduino-cli/0.34/platform-specification/#platformlocaltxt
https://arduino.github.io/arduino-cli/0.34/commands/arduino-cli_compile/
